### DISCUSSION TITLE
@click on submit button prevents submission in Chromium browsers.

### DISCUSSION THREAD

**Discussion by ghost** [alpinejs/alpine#2414] [7. Bugs (archived)] [answered]
```html
<div class="w-full" x-data="{spinner: false}">
   <button type="submit" @click="spinner =! spinner;" x-bind:disabled="spinner" class="inline-flex items-center justify-center bg-sky-700 text-white hover:bg-sky-600 disabled:bg-gray-500 mt-8 w-full py-3 px-6 border border-transparent rounded-md font-medium">
      <svg class="animate-spin -ml-1 mr-3 h-5 w-5 text-white" x-show="spinner" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24">
         <circle class="opacity-25" cx="12" cy="12" r="10" stroke="currentColor" stroke-width="4"></circle>
         <path class="opacity-75" fill="currentColor" d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z">
         </path>
      </svg>
      <span x-show="!spinner">Place Order</span>
      <span x-show="spinner">Loading ...</span>
   </button>
</div>
```

This works fine in Firefox. It disables the button and shows the spinner and Loading... text. In Chromium browsers it just disables the button, shows the text but never submits the form. Nothing happens in Devtools.

I can make it work by adding `document.forms['paymentform'].submit();` to the @click but I don't think that's how it should be? It's like Alpine does a preventDefault in Chromium.

JSFiddle: https://jsfiddle.net/cnfgst42/ (Works in Firefox, not in Chrome/Brave/Safari etc.)

**Comment by SimoTod** [comment, marked as answer]
I believe it's related to what Alpine returns when it evaluates the string.
Returning false on a click handler on a submit button, prevent the form submissions in some browsers.
It seems to work okay if you use something like
```html
<div class="w-full" x-data="{spinner: false, submit() {spinner = true}}">
   <button type="submit" @click="submit()" x-bind:disabled="spinner" ...
```
